internal/consensus/state.go

```diff
@@ -1049,7 +1049,7 @@ func (cs *State) handleMsg(ctx context.Context, mi msgInfo, fsyncUponCompletion
 		// once proposal is set, we can receive block parts
 		err = cs.setProposal(msg.Proposal, mi.ReceiveTime)
 		// See if we can try creating the proposal block if keys exist
-		if err != nil && cs.config.GossipTransactionKeyOnly && cs.privValidatorPubKey != nil {
+		if err == nil && cs.config.GossipTransactionKeyOnly && cs.privValidatorPubKey != nil {
```

Comment:
Oh, interesting... 

Could we add a unit test for this to assert if Tx key Dissemination was used? Could even just refactor out this snippet to be a helper if that makes it easier

Reply:
Yeah +1 to maybe find a way to test if gossip is on or not under given condition